### PR TITLE
Added a post-processing step to the XAML generation

### PR DESCRIPTION
**Pull request by WKleinschmit**
In order to get rid of some of the unneccessary verbosity.
Give it a try and tell me what you think.

### CODE REVIEW

File: MainWindow.xaml.cs [hunk 1]
```diff
@@ -25,14 +26,14 @@ public MainWindow()
             InitializeComponent();
 
             // get syntax coloring styles
-            _bracketStyle   = (Style)Resources["BracketStyle"];
-            _elementStyle   = (Style)Resources["ElementStyle"];
-            _quotesStyle    = (Style)Resources["QuotesStyle"];
-            _textStyle      = (Style)Resources["TextStyle"];
+            _bracketStyle = (Style)Resources["BracketStyle"];
```
Comment (WKleinschmit) [inline MainWindow.xaml.cs:29]: Sorry. Most of these changes are ReSharper™ re-formats.

File: MainWindow.xaml.cs [hunk 2]
```diff
@@ -231,7 +235,10 @@ private FlowDocument CreateFlowDocument(bool success, string serializedStyle)
                                     }
 
                                     paragraph.AddRun(_quotesStyle, "\"");
+                                    paragraph.AddLineBreak();
```
Comment (WKleinschmit) [inline MainWindow.xaml.cs:238]: I like to have each xml attribute on it's own line.
It greatly improves readabillity.

File: MainWindow.xaml.cs [hunk 3]
```diff
@@ -231,7 +235,10 @@ private FlowDocument CreateFlowDocument(bool success, string serializedStyle)
...
+                                    paragraph.AddRun(_textStyle, new string(' ', indent * 4 + elementName.Length + 1));
                                 }
+                                paragraph.RemoveLastLineBreak();
```
Comment (WKleinschmit) [inline MainWindow.xaml.cs:244]: Have to do this or else the closing bracket will also be on its own line.

File: MainWindow.xaml.cs [hunk 4]
```diff
@@ -231,7 +235,10 @@ private FlowDocument CreateFlowDocument(bool success, string serializedStyle)
                                     }
 
                                     paragraph.AddRun(_quotesStyle, "\"");
+                                    paragraph.AddLineBreak();
+                                    paragraph.AddRun(_textStyle, new string(' ', indent * 4 + elementName.Length + 1));
```
Comment (WKleinschmit) [inline MainWindow.xaml.cs:242]: I like to have each xml attribute on it's own line.
It greatly improves readabillity.

File: MainWindow.xaml.cs [hunk 5]
```diff
@@ -301,5 +308,80 @@ private void OnPropertyChanged([CallerMemberName] string propertyName = null)
...
+                            {
+                                elt.SetAttributeValue("Value", value);
+                                eltValueNode.Remove();
+                            }
+                            break;
+                        }
+                }
+
+            }
+        }
```
Comment (WKleinschmit) [inline MainWindow.xaml.cs]: Her is the fun part ;-)

File: MainWindow.xaml.cs [hunk 6]
```diff
@@ -144,8 +145,10 @@ private void ShowStyle(object sender, SelectionChangedEventArgs e)
             {
                 var success = TrySerializeStyle(style.ResourceKey, out var serializedStyle);
 
+                var styleXml = CleanupStyle(serializedStyle);
+
```
Comment (WKleinschmit) [inline MainWindow.xaml.cs]: See the fun part at the end...

**Comment by drewnoakes** [comment]
Hey @WKleinschmit this looks great, thanks! I've wanted to tidy up the output for a long time. I made some baby steps recently, but this is a great improvement.

There are some further clean ups we could do. I'll file some issues to track them and tag you FYI.

Can you confirm where the icon came from and that you have the right to use it here?

**Comment by drewnoakes** [comment]
I pushed some more commits to this PR. I am happy to merge this and make a new release once you confirm that the project has the right to use the icon.

**Comment by WKleinschmit** [comment]
Hello,
The icon is from this site: https://thenounproject.com/term/nose/
It is Creative Commons and I modified it a bit to make it work in an 16x16 icon.
So, it should be fine.

**Comment by drewnoakes** [comment]
Most of these icons are CCBY, meaning we have to give attribution. I cannot tell which of these icons you derived your icon from, so I do not know who to give attribution to.

I will remove that commit from the PR and merge it. Feel free to submit another PR that includes the icon and attribution in the README.

**Pull request merged**